src/enum_net.cpp

```diff
@@ -173,6 +173,52 @@ namespace {
 		);
 	}
 
+#if TORRENT_USE_GETIPFORWARDTABLE || TORRENT_USE_NETLINK
+	address build_netmask(int bits, int family)
+	{
+		if (family == AF_INET)
+		{
+			using bytes_t = boost::asio::ip::address_v4::bytes_type;
+			bytes_t b;
+			std::memset(&b[0], 0xff, b.size());
+			for (int i = int(b.size()) - 1; i > 0; --i)
+			{
+				if (bits < 8)
+				{
+					b[i] <<= bits;
+					break;
+				}
+				b[i] = 0;
+				bits -= 8;
+			}
+			return address_v4(b);
+		}
+#if TORRENT_USE_IPV6
+		else if (family == AF_INET6)
+		{
+			using bytes_t = boost::asio::ip::address_v6::bytes_type;
+			bytes_t b;
+			std::memset(&b[0], 0xff, b.size());
```

Comment:
are we guaranteed that the bytes type is a `std::array`? if so you could use `b.fill(0)` (with the main benefit of not using `memset()`)

Reply:
I'm also thinking. It feels like there should be a simpler way of generating these bitmasks. stackoverflow suggests a table of 32 entries. But that won't be quite as nice for IPv6. It seems at least that the core of this algorithm could be factored out, and operate on a span.

the `sha1_hash` has function that do things like this too. I'm not sure if they are trivially reusable though. I suppose that's more plain shifting, so a little bit more generic.